src/sentry/static/sentry/app/components/events/eventCause.jsx

```diff
@@ -63,14 +63,16 @@ export default createReactClass({
     }
 
     let commits = [];
+    let commitSet = new Set();
```

Comment:
I think you should use https://lodash.com/docs/4.17.5#uniqBy
```
let { commiters } = this.state;

let commitsWithAuthors = _.flatMap(committers, ({ commits, author }) =>
  commits.map(commit => ({
    ...commit,
    author: author
  }))
);

let uniqueCommitsWithAuthors = _.uniqBy(
  commitsWithAuthors,
  commit => commit.id
);

```

Reply:
then i have to read documentation on wtf a flatMap is and what a uniqBy is

why cant the precompilers optimize this instead of me importing obscure functions

Reply:
also ill note that your code would be slower given that it duplicates the commit objects whereas my optimizes out the data duplication

Reply:
split the difference, push the logic into a function that explains what it does and so people don't have to read it in the render method?

Reply:
not going to make changes at this point as it increases the time to complete these fixes and doesnt provide any more simplicity than what is already present